### PR TITLE
add volume storage

### PR DESCRIPTION
**Pull request by AerysNan**

### CODE REVIEW

**Comment by AerysNan** [comment]
LGTM

File: storage/handler.go [hunk 1]
```diff
@@ -2,83 +2,165 @@ package storage
...
+	name := path.Join(s.root, fmt.Sprintf("%d.dat", s.currentVolume.volumeId))
+	file, err := os.OpenFile(name, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0766)
+	if err != nil {
+		logrus.WithError(err).Errorf("Open file %v failed", name)
+		return nil, osserror.ErrServerInternal
+	}
+	defer file.Close()
+	bytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(bytes, uint64(size))
+	_, err = file.Write(bytes)
```
Comment (OneSizeFitsQuorum) [inline storage/handler.go:146]: 这里应该用WriteAt(bytes,s.currentVolume.size)吧

File: storage/handler.go [hunk 2]
```diff
@@ -2,83 +2,165 @@ package storage
...
 	if err != nil {
-		return nil, err
+		logrus.WithError(err).Errorf("Write file %v failed", name)
+		return nil, osserror.ErrServerInternal
 	}
-	_, err = file.WriteString(request.Body)
+	_, err = file.Write([]byte(data))
```
Comment (OneSizeFitsQuorum) [inline storage/handler.go:151]: 同上

File: storage/handler.go [hunk 3]
```diff
@@ -2,83 +2,165 @@ package storage
...
 func NewStorageServer(address string, root string, metadataClient pm.MetadataForStorageClient) *StorageServer {
 	storageServer := &StorageServer{
 		metadataClient: metadataClient,
 		address:        address,
 		root:           root,
+		volumes:        make(map[int64]*Volume),
+		currentVolume:  new(Volume),
 	}
-	go storageServer.heartbeatLoop()
+	storageServer.recover()
```
Comment (OneSizeFitsQuorum) [inline storage/handler.go:51]: 这里recover应该可以异步吧？

Comment (AerysNan) [reply storage/handler.go:51]: won't fix

File: storage/handler.go (hunk 1 above)
Comment (AerysNan) [reply storage/handler.go:146]: won't fix

File: storage/handler.go (hunk 2 above)
Comment (AerysNan) [reply storage/handler.go:151]: won't fix

**Pull request merged**